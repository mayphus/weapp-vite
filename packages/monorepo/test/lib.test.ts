import { setPkgJson } from '@/lib'

describe('lib', () => {
  it('setPkgJson casae 0', () => {
    const t = {}
    const o = {}
    setPkgJson(t, o)
    expect(o).toMatchSnapshot()
  })

  it('setPkgJson casae 1', () => {
    const t = {

    }
    const o = {
      scripts: {
        'build': 'turbo run build',
        'dev': 'turbo run dev --parallel',
        'test': 'vitest run --coverage.enabled',
        'test:dev': 'vitest',
        'lint': 'turbo run lint',
        'release': 'changeset',
        'publish-packages': 'turbo run build lint test && changeset version && changeset publish',
        'preinstall': 'npx only-allow pnpm',
        'prepare': 'husky',
        'commit': 'commit',
      },
    }
    setPkgJson(t, o)
    expect(o).toMatchSnapshot()
  })
})
