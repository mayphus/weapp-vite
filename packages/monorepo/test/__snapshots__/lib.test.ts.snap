// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lib > setPkgJson casae 0 1`] = `
{
  "packageManager": "",
  "scripts": {
    "script:clean": "tsx scripts/monorepo/clean.ts",
    "script:init": "tsx scripts/monorepo/init.ts",
    "script:mirror": "tsx scripts/monorepo/binaryMirror.ts",
    "script:sync": "tsx scripts/monorepo/sync.ts",
  },
}
`;

exports[`lib > setPkgJson casae 1 1`] = `
{
  "packageManager": "",
  "scripts": {
    "build": "turbo run build",
    "commit": "commit",
    "dev": "turbo run dev --parallel",
    "lint": "turbo run lint",
    "preinstall": "npx only-allow pnpm",
    "prepare": "husky",
    "publish-packages": "turbo run build lint test && changeset version && changeset publish",
    "release": "changeset",
    "script:clean": "tsx scripts/monorepo/clean.ts",
    "script:init": "tsx scripts/monorepo/init.ts",
    "script:mirror": "tsx scripts/monorepo/binaryMirror.ts",
    "script:sync": "tsx scripts/monorepo/sync.ts",
    "test": "vitest run --coverage.enabled",
    "test:dev": "vitest",
  },
}
`;
